src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassReportServiceImpl.java

```diff
@@ -0,0 +1,302 @@
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Proposal> proposalResultGrouping = processingResults.getProposalResults();
+        List<RassBusinessObjectUpdateResult<Proposal>> proposalResults = proposalResultGrouping.getObjectResults();
+        writeBOResultsSummaryToReport(proposalResults, Proposal.class, false);
+    }
+
+    private void writeAwardSummaryToReport(RassBatchJobReport rassBatchJobReport) {
+        RassXmlProcessingResults processingResults = rassBatchJobReport.getProcessingResults();
+        RassBusinessObjectUpdateResultGrouping<Award> awardResultGrouping = processingResults.getAwardResults();
+        List<RassBusinessObjectUpdateResult<Award>> awardResults = awardResultGrouping.getObjectResults();
+        writeBOResultsSummaryToReport(awardResults, Award.class, true);
+    }
+
+    private <R extends PersistableBusinessObject> void writeBOResultsSummaryToReport(List<RassBusinessObjectUpdateResult<R>> results, Class businessObjectClass,
+            boolean shouldReportUpdates) {
+        writeResultsSummaryToReportByResultCode(results, RassConstants.RassObjectUpdateResultCode.SUCCESS_NEW, businessObjectClass);
+        if (shouldReportUpdates) {
+            writeResultsSummaryToReportByResultCode(results, RassConstants.RassObjectUpdateResultCode.SUCCESS_EDIT, businessObjectClass);
+        }
+        writeResultsSummaryToReportByResultCode(results, RassConstants.RassObjectUpdateResultCode.ERROR, businessObjectClass);
+        writeResultsSummaryToReportByResultCode(results, RassConstants.RassObjectUpdateResultCode.SKIPPED, businessObjectClass);
+    }
+
+    private <R extends PersistableBusinessObject> void writeResultsSummaryToReportByResultCode(List<RassBusinessObjectUpdateResult<R>> results,
+            RassObjectUpdateResultCode resultCode, Class businessObjectClass) {
+        List<RassBusinessObjectUpdateResult<R>> filteresdResults = results.stream().filter(result -> resultCode.equals(result.getResultCode()))
+                .collect(Collectors.toList());
+        writeSummaryLineToReport(filteresdResults, resultCode, businessObjectClass);
+    }
+
+    private <R extends PersistableBusinessObject> void writeSummaryLineToReport(List<RassBusinessObjectUpdateResult<R>> filteresdResults,
+            RassObjectUpdateResultCode resultCode, Class businessObjectClass) {
+
+        String resultMessage = StringUtils.EMPTY;
+        switch (resultCode) {
+            case SUCCESS_NEW:
+                resultMessage = " created";
+                break;
+            case SUCCESS_EDIT:
+                resultMessage = " updated";
+                break;
+            case ERROR:
+                resultMessage = " that resulted in error";
+                break;
+            case SKIPPED:
+                resultMessage = " skipped";
+                break;
+        }
```

Comment:
It might be a good idea to add a "default" case to this switch block.

Reply:
added